src/OpenSage.Game/Logic/Player.cs

```diff
@@ -38,6 +38,21 @@ public int GetEnergy(GameObjectCollection allGameObjects)
             return energy;
         }
 
+        public void SpendMoney(uint amount)
+        {
+            if (Money >= amount)
+            {
+                Money -= amount;
+            }
+            else
+            {
+                // this should not happen since we should check first if we can spend that much
+                Money = 0;
```

Comment:
We should probably log something here atleast

Reply:
I agree, will do.

Reply:
done